drivers/interrupt_controller/Kconfig.vim

```diff
@@ -0,0 +1,15 @@
+# Copyright (C) 2023 BeagleBoard.org Foundation
+# Copyright (C) 2023 S Prashanth
+#
+# SPDX-License-Identifier: Apache-2.0
+
+if CPU_CORTEX
+
+config VIM
+	bool "Vectored Interrupt Manager"
+	depends on ARM_CUSTOM_INTERRUPT_CONTROLLER
+	help
+		The Vectored Interrupt Manager provides hardware assistance for prioritizing
+		and controlling the many interrupt sources for ARM Cortex-family processors.
+
```

Comment:
this should be `depends on DT_HAS_ARM_VIM_ENABLED`. Check how this is done for the other interrupt controllers.